rewriter/Prop.cc

```diff
@@ -240,8 +302,21 @@ vector<unique_ptr<ast::Expression>> processProp(core::MutableContext ctx, const
                                                      ASTUtil::dupType(getType.get()));
         auto insSeq = ast::MK::InsSeq1(loc, std::move(assertTypeMatches), ast::MK::RaiseUnimplemented(loc));
         nodes.emplace_back(ASTUtil::mkGet(ctx, loc, name, std::move(insSeq)));
-    } else if (ret.ifunset == nullptr && forTStruct) {
-        nodes.emplace_back(ASTUtil::mkGet(ctx, loc, name, ast::MK::Instance(nameLoc, ivarName)));
+    } else if (ret.ifunset == nullptr) {
+        if (knownNonModel(syntacticSuperClass)) {
+            if (wantTypedInitialize(syntacticSuperClass)) {
+                nodes.emplace_back(ASTUtil::mkGet(ctx, loc, name, ast::MK::Instance(nameLoc, ivarName)));
+            } else {
+                // Need to hide the instance variable access, because there wasn't a typed constructor to declare it
+                auto castSelf = ast::MK::Unsafe(loc, ast::MK::Self(loc));
+                auto ivarGet = ast::MK::Send1(loc, std::move(castSelf), core::Names::instanceVariableGet(),
+                                              ast::MK::Symbol(nameLoc, ivarName));
+                nodes.emplace_back(ASTUtil::mkGet(ctx, loc, name, std::move(ivarGet)));
+            }
+        } else {
+            // Models have a custom decorator, which means we have to forward the prop get to it.
+            nodes.emplace_back(ASTUtil::mkGet(ctx, loc, name, ast::MK::RaiseUnimplemented(loc)));
+        }
```

Comment:
This is the first half of the change. This basically means that actually make getters for "known non-model" subclasses. That means T::Struct, T::InexactStruct, and Chalk::ODM::Document when we see those as the syntactic super class.

For all other cases (i.e., any other syntactic super class), we leave the body as a `raise`.